### DISCUSSION TITLE
[WIP] changes for next release 3.10.4

### DISCUSSION THREAD

**Discussion by nlohmann** [nlohmann/json#3064] [Announcements]
This thread collects all changes made since the 3.10.3 (#3065) release.

### Bug fixes

- Fix regression bug introduced in release 3.10.0 which broke compilation for types with an explicit default constructor with default arguments.
  - #3077
  - #3079
- Fix regression bug introduced in release 3.10.0 which treated the return values of `std::find` and `std::remove` as pointers which could break compilation.
  - #3081 
  - #3082
- Fix converting `std::filesystem::path` to JSON. Before release 3.10.3, such a conversion would trigger a stack overflow. Release 3.10.3 then further broke compilation on Windows. By @theodelrieu.
  - #3070
  - #3073

**Comment by theodelrieu** [comment]
3.10.3 did not introduce the stack overflow on Windows, it made it stop compiling.

**Reply by nlohmann** [reply]
Oh, I got confused there. So the correct wording would be:

> Fix regression bug introduced in release 3.10.3 which broke compilation of converting std::filesystem::path to JSON on Windows.

**Reply by theodelrieu** [reply]
And it fixes the stack overflow as well 🙂

**Reply by nlohmann** [reply]
Now I'm even more confused... so we had a stack overflow on non-Windows systems?

**Reply by theodelrieu** [reply]
On Windows only.

**Reply by nlohmann** [reply]
So we had a stack overflow prior to 3.10.3, then 3.10.3 broke compilation, and now everything is fine?

**Reply by theodelrieu** [reply]
Yes
